File: Vohra_ProMongoDB_SourceCode/ch4/mongodbscripts/catalog.rb
require "mongo_mapper"

class Catalog
  include MongoMapper::Document
   
MongoMapper.connection = Mongo::Connection.new('localhost', 27017)
MongoMapper.database = "local"

  key :catalogId, String
  key :journal, String
  key :publisher, String
  key :edition, String
  key :title, String
  key :author, String
 
end
 
 
 
catalog = Catalog.create({
  :catalogId => 'catalog1',
  :journal => 'Oracle Magazine',
  :publisher => 'Oracle Publishing',
  :edition => 'November December 2013',
  :title => 'Engineering as a Service',
  :author => 'David A. Kelly'
})
 
catalog.save

catalog = Catalog.create({
  :catalogId => 'catalog2',
  :journal => 'Oracle Magazine',
  :publisher => 'Oracle Publishing',
  :edition => 'November December 2013',
  :title => 'Quintessential and Collaborative',
  :author => 'Tom Haunert'
})
 
catalog.save
 
##print Catalog.find(:first)
print "\n"
##print Catalog.find(:all)
 

File: Vohra_ProMongoDB_SourceCode/ch1/MongoDBJava/src/mongodb/Catalog.java
package mongodb;

import java.util.Map;
import java.util.Set;
import java.io.Serializable;
import org.bson.BSONObject;

import com.mongodb.BasicDBObject;
import com.mongodb.DBObject;

public class Catalog extends BasicDBObject implements Serializable {

	public static final long serialVersionUID = 1L;

	public String catalogId;
	public String journal;
	public String publisher;
	public String edition;
	public String title;
	public String author;

	public Catalog() {
		super();
	}

	public Catalog(String catalogId, String journal, String publisher,
			String edition, String title, String author) {
		this.catalogId = catalogId;
		this.journal = journal;
		this.publisher = publisher;
		this.edition = edition;
		this.title = title;
		this.author = author;
	}
}
